tests/accelerators/test_ddp.py

```diff
@@ -102,3 +103,28 @@ def test_torch_distributed_backend_env_variables(tmpdir):
                 logger=False,
             )
             trainer.fit(model)
+
+
+@pytest.mark.parametrize('move_to_device_pre_dispatch_enabled', [True, False])
+@mock.patch('pytorch_lightning.plugins.DDPPlugin.model_to_device')
+def test_move_to_device_in_pre_dispatch(mock_model_to_device, tmpdir, move_to_device_pre_dispatch_enabled):
```

Comment:
Two things:

- Would've liked to add a spawn test but mocks don't seem to carry across to new processes?
- Can I combine patch and parametrize?

Reply:
1. Nope, you'd have to create the mock in each process
2. What do you mean? Having different mocks for each parametrize? I don't think so

Reply:
You might be able to do so with a callback, applying the patch context manager.

Reply:
You can make patch conditional on parametrize. with the return object.